BaseLib/ConfigTree-impl.h

```diff
@@ -50,15 +52,40 @@ getConfParam(std::string const& param, T const& default_value) const
     return default_value;
 }
 
-template<typename T>
-boost::optional<T>
-ConfigTree::
-getConfParamOptional(std::string const& param) const
+template <typename T>
+boost::optional<T> ConfigTree::getConfParamOptional(
+    std::string const& param) const
 {
     checkUnique(param);
 
+    return getConfParamOptionalImpl(param, static_cast<T*>(nullptr));
+}
+
+template <typename T>
+boost::optional<T> ConfigTree::getConfParamOptionalImpl(
+    std::string const& param, T*) const
+{
+    if (auto p = getConfSubtreeOptional(param)) return p->getValue<T>();
+
+    return boost::none;
+}
+
+template <typename T>
+boost::optional<std::vector<T>> ConfigTree::getConfParamOptionalImpl(
+    std::string const& param, std::vector<T>*) const
+{
     if (auto p = getConfSubtreeOptional(param))
-        return p->getValue<T>();
+    {
+        std::stringstream ss{p->getValue<std::string>()};
+        std::vector<T> result;
+        T value;
+        while (ss >> value)
+            result.push_back(value);
+        if (!ss.eof()) // The stream is not read until the end, must be an error
```

Comment:
I see you have some error checking here, I overlooked that last week. The problem with your solution is, that `boost::none` in the default case means "parameter has not been found"—cf. the implementation of `getConfParam<>()`—, i.e., it probably triggers confusing error messages.
Thus, you should print a proper error message here and abort rather than returning `none`. I think it would be good if that error message included the index of the value at which the error occured:
```
error("Value for key <" + param + "> `" + shortString(p->data())
+ "' not convertible to a vector of the desired type."
" Could not convert token no. " + std::to_string(result.size()+1) + ".");
```